conda_lock/lockfile/__init__.py

```diff
@@ -207,13 +207,22 @@ def write_section(text: str) -> None:
                 """
             )
 
-        yaml.dump(
-            {
-                "version": Lockfile.version,
-                **json.loads(
-                    content.json(by_alias=True, exclude_unset=True, exclude_none=True)
-                ),
-            },
-            stream=f,
-            sort_keys=False,
-        )
+        output: Dict[str, Any] = {
+            "version": Lockfile.version,
+            "metadata": json.loads(
+                content.metadata.json(
```

Comment:
I just noticed that in #390 you have here instead `"metadata": content.metadata.dict(` which seems more elegant. Is this on purpose, or does this cover some edge case?

Reply:
I believe there was an issue when encoding `content.metadata` as a dict due to a special class / container that was used in the metadata (was it a frozenset?). Let me check for certain.